### PR TITLE
Add annotation reader cache and fix doctrine typeGuesser

### PR DESCRIPTION
**Pull request by Damienbelingheri**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes
| Documented?   | no
| Fixed tickets | #818 & #856
| License       | MIT

Add an cache for annotations reader to improve performances.
And fix doctrine typeGuesser with doctrine 2.9

### CODE REVIEW

**Review by Vincz — DISMISSED** [review]

File: docs/annotations/index.md [hunk 1]
```diff
@@ -13,7 +15,7 @@ overblog_graphql:
           dir: "%kernel.project_dir%/src/GraphQL"
           suffix: ~
 ```
-
+To use annoation, you must instal   '  '' ' and use the annatotton mapping type
```
Comment (mcg-web) [inline docs/annotations/index.md:18]: remove this please

File: composer.json [hunk 2]
```diff
@@ -3,7 +3,10 @@
     "type": "symfony-bundle",
     "license": "MIT",
     "description": "This bundle provides tools to build a GraphQL server in your Symfony App.",
-    "keywords": ["GraphQL", "Relay"],
+    "keywords": [
```
Comment (mcg-web) [inline composer.json:6]: revert all changes on this file please.

File: src/Config/Parser/AnnotationParser.php [hunk 3]
```diff
@@ -32,18 +36,34 @@ protected static function getMetadatas(Reflector $reflector): array
...
-                !class_exists(AnnotationRegistry::class)) {
+        if (!isset(self::$annotationReader)) {
+            if (!class_exists(AnnotationReader::class) || !class_exists(AnnotationRegistry::class)) {
                 // @codeCoverageIgnoreStart
-                throw new RuntimeException('In order to use graphql annotation, you need to require doctrine annotations');
+                throw new RuntimeException("In order to use annotations, you need to install 'doctrine/annotations' first. See: 'https://www.doctrine-project.org/projects/annotations.html'");
                 // @codeCoverageIgnoreEnd
             }
+            if (!class_exists(ApcuAdapter::class)) {
+                // @codeCoverageIgnoreStart
```
Comment (mcg-web) [inline src/Config/Parser/AnnotationParser.php:48]: we ignore a lot of code here can we add tests to cover these intead ?

**Review by Vincz — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,10 +1,10 @@
 OverblogGraphQLBundle
 ======================
 
-![CI](https://github.com/overblog/GraphQLBundle/workflows/CI/badge.svg?branch=0.14)
-[![Build status](https://ci.appveyor.com/api/projects/status/7ksxlcgwt40q74hv/branch/0.14?svg=true)](https://ci.appveyor.com/project/overblog/graphqlbundle/branch/0.14)
-[![Scrutinizer Code Quality](https://scrutinizer-ci.com/g/overblog/GraphQLBundle/badges/quality-score.png?b=0.14)](https://scrutinizer-ci.com/g/overblog/GraphQLBundle/?branch=0.14)
-[![Coverage Status](https://coveralls.io/repos/github/overblog/GraphQLBundle/badge.svg?branch=0.14)](https://coveralls.io/github/overblog/GraphQLBundle?branch=0.14)
+![CI](https://github.com/overblog/GraphQLBundle/workflows/CI/badge.svg?branch=master)
```
Comment (mcg-web) [inline README.md:4]: Revert please

File: composer.json [hunk 5]
```diff
@@ -68,15 +70,15 @@
     },
     "extra": {
         "branch-alias": {
-            "dev-master": "0.14-dev"
+            "dev-master": "1.0-dev"
```
Comment (mcg-web) [inline composer.json:73]: Revert please

**Review by mcg-web — CHANGES_REQUESTED** [review]
Look good to me, can you just revert master changes from 0.14 branch please, before merging this ?

**Review by mcg-web — APPROVED** [review]

**Pull request merged**